scheduler.go

```diff
@@ -473,6 +473,10 @@ func (s *scheduler) selectStart() {
 				next = j.next(s.now())
 			}
 
+			if j.stopTimeReached() {
```

Comment:
This check shouldn't need to happen here, but instead in the `selectExecJobsOutForRescheduling` function. since the executor is checking, if the stop time is reached, new jobs will be caught by that and then not rescheduled.